calyx/src/analysis/live_range_analysis.rs

```diff
@@ -612,8 +612,32 @@ impl LiveRangeAnalysis {
         invoke: &ir::Invoke,
         shareable_components: &ShareSet,
     ) -> (TypeNameSet, TypeNameSet) {
-        //The reads of the invoke include its inputs plus the cell itself, if the
-        //outputs are not empty.
+        let mut comp_is_written = false;
+
+        //The writes of the invoke include its outpus plus the cell itself, if the
+        //inputs are not empty.
```

Comment:
Can you explain why the inputs need to be non-empty in this case?

Reply:
That comment is kind of ambiguous. I meant to say that the inputs not being empty is only a requirement for the "cell itself" (i.e., the cell being invoked) being counted as a write.

Reply:
The way I interpreted the comment was "this component can only write if it has 1 or more input cells." I think this interpretation is incorrect though.

Reply:
Got it, thanks. It might be helpful for future readers to add this clarification to the comment